packages/api/src/libs/osnap/utils.ts

```diff
@@ -0,0 +1,28 @@
+import assert from "assert";
+import { Json } from "../../types";
+import type { OsnapPluginData, Transaction } from "./interfaces";
+import { simulateTenderlyTx, TenderlySimulationParams } from "@uma/common";
+
+export async function simulateOsnapTx(space: OsnapPluginData, transactionIndex = 0): Promise<Json> {
+  const { safe } = space;
+  assert(safe, "Requires a safe");
+  const { network, transactions } = safe;
+  assert(transactions.length > transactionIndex, "Cannot simulate not existent transaction");
+
+  // we only expect a single tx
```

Comment:
I don't think so. This would be the case if Snapshot frontend was to multicall batched transactions (like it did for the old safeSnap plugin). But oSnap plugin relies on oSnap module contract to handle batching, so oSnap proposal accepts array of tx to execute

Reply:
we have restricted the new osnap plugin to only do a single transaction, so thats what i am basing this on. the old osnap tx builder will not be using this feature